File: WebApp.Core/Entities/Compensation/BusinessRule.cs
// --------------------------------------------------------------------------------------------------------------------
// <copyright file="BusinessRule.cs" company="">
//
// </copyright>
// <summary>
//   Defines the BusinessRule type.
// </summary>
// --------------------------------------------------------------------------------------------------------------------

using System;
using System.Collections.Generic;
using WebApp.Core.Entities.Configuration;

namespace WebApp.Core.Entities.Compensation
{
    public class BusinessRule : IEntityBase
  {
    public int Id { get; set; }

    public int Version { get; set; }

    public DateTime StartDate { get; set; }

    public int IncidentTypeId { get; set; }

    public CatalogItem IncidentType { get; set; }

    public int? PaymentTableId { get; set; }

    public PaymentTable PaymentTable { get; set; }

    public List<BusinessRulePaymentVariableElement> PaymentVariableElements { get; set; }

    public bool Additional { get; set; }

    public decimal? SalaryPercentage { get; set; }
  }
}

File: BaseWebApp/ClientApp/src/app/core/models/compensation/BusinessRule.ts
import { IPaymentTable } from '@app/core';

export interface IBusinessRule {
    id: number;
    startDate: Date;
    incidentTypeId: number;
    paymentTableId: number;
    paymentTable: IPaymentTable;
    version: number;
    salaryPercentage: number;
    additional: boolean;
    paymentVariableElements: Array<{ id: string, itemName: string }>;
}
